### PR TITLE
Implement CSR resource.

### PR DESCRIPTION
**Pull request by dimitarproynov**
**Summary of Pull Request**

The resource export the CSR in encoded (PEM) and decoded (plain string) formats, so that users can copy-paste the CSR content in external CA.

**Type of Pull Request**

- [ ] This is a bug fix.
- [X] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [X] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

For bug fixes or features:

- [X] Tests have been completed.
- [X] Documentation has been added/updated.

make build
make lint
make test

=== RUN   TestAccResourceVcfCsr
=== PAUSE TestAccResourceVcfCsr
=== CONT  TestAccResourceVcfCsr
--- PASS: TestAccResourceVcfCsr (47.41s)
PASS

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

**Review by tenthirtyam — APPROVED** [review]
LGTM!

File: internal/provider/resource_csrs.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if resourceFqdn == nil {
+		return diag.FromErr(fmt.Errorf("could not determine FQDN for resourceType %s in domain %s", resourceType, domainId))
+	}
+
+	country := data.Get("country").(string)
+	email := data.Get("email").(string)
+	keySize := strconv.Itoa(data.Get("key_size").(int))
```
Comment (vasilsatanasov) [inline internal/provider/resource_csrs.go:122]: Wouldn't data.Get("key_size").(string) do the same ? Just speculating here.

**Review by vasilsatanasov — APPROVED** [review]
LGTM!

File: internal/provider/resource_csrs.go (hunk 1 above)
Comment (dimitarproynov) [reply internal/provider/resource_csrs.go:122]: panic: interface conversion: interface {} is int, not string

**Pull request merged**